File: client/src/api/restaurants/RestaurantDto.ts
import { CuisineDto } from "../cuisines/CuisineDto";
import { MenuDto } from "../menu/MenuDto";
import { OpeningTimes } from "./OpeningTimes";

export interface RestaurantDto {
  id: string;
  managerId: string;
  name: string;
  description?: string;
  phoneNumber: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  postcode: string;
  latitude: number;
  longitude: number;
  status: RestaurantStatus;
  openingTimes: OpeningTimes;
  deliveryFee: number;
  minimumDeliverySpend: number;
  maxDeliveryDistanceInKm: number;
  estimatedDeliveryTimeInMinutes: number;
  menu: MenuDto;
  cuisines: CuisineDto[];
}

type RestaurantStatus = "PendingApproval" | "Approved";

File: server/src/Web/Features/Restaurants/RestaurantDto.cs
using System;
using System.Collections.Generic;
using Web.Features.Cuisines;
using Web.Features.Menus;

namespace Web.Features.Restaurants
{
    public record RestaurantDto
    {
        public Guid Id { get; init; }
        public Guid ManagerId { get; init; }
        public string Name { get; init; }
        public string PhoneNumber { get; init; }
        public string Address { get; init; }
        public float Latitude { get; init; }
        public float Longitude { get; init; }
        public string Status { get; init; }
        public OpeningTimesDto OpeningTimes { get; init; } = new();
        public decimal DeliveryFee { get; init; }
        public decimal MinimumDeliverySpend { get; init; }
        public int MaxDeliveryDistanceInKm { get; init; }
        public int EstimatedDeliveryTimeInMinutes { get; init; }
        public List<CuisineDto> Cuisines { get; init; } = new();
        public MenuDto Menu { get; init; }
    }
}
